apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsFilters/AlertsFilters.tsx

```diff
@@ -28,62 +32,74 @@ export const AlertsFilters: FunctionComponent<{
       })) ?? [],
     [],
   );
-  const filters = [
-    {
-      title: 'Assignee',
-      accessor: 'assigneeId',
-      options: [
-        {
-          label: 'Unassigned',
-          value: null,
+  const filters: IFilterDefinition[] = useMemo(
+    () => [
+      {
+        id: 'assigneeId',
+        label: 'Assignee',
+        type: 'multi-select',
+        params: {
+          options: [
+            {
+              label: 'Unassigned',
+              value: null,
+            },
+            ...assigneeOptions,
+          ],
         },
-        ...assigneeOptions,
-      ],
-    },
-    {
-      title: 'Status',
-      accessor: 'status',
-      options: statusOptions,
-    },
-    {
-      title: 'Label',
-      accessor: 'label',
-      options: labels.map(label => ({
-        label,
-        value: label,
-      })),
-    },
-  ] satisfies Array<{
-    title: string;
-    accessor: string;
-    options: Array<{
-      label: string;
-      value: string | null;
-    }>;
-  }>;
+      },
+      {
+        id: 'status',
+        label: 'Status',
+        type: 'multi-select',
+        params: {
+          options: statusOptions,
+        },
+      },
+      {
+        label: 'Label',
+        id: 'label',
+        type: 'multi-select',
+        params: {
+          options: labels.map(label => ({
+            label,
+            value: label,
+          })),
+        },
+      },
+    ],
+    [assigneeOptions, labels, statusOptions],
+  );
+  console.log('filters', filters);
   const { filter, onFilter } = useFilter();
+  const filterValues = useMemo(
+    () => Object.entries(filter || {}).map(([key, value]) => ({ id: key, value })),
+    [filter],
+  );
+
   const onClearSelect = useCallback(
-    (accessor: string) => () => {
-      onFilter(accessor)([]);
+    (filterId: TFiltersInputID) => () => {
+      onFilter(filterId)([]);
+    },
+    [onFilter],
+  );
+
+  const onFilterChange = useCallback(
+    (filterValue: IFilterValue) => {
+      onFilter(filterValue.id)(filterValue.value as Array<string | null>);
     },
     [onFilter],
   );
 
   return (
-    <div>
-      <h4 className={'leading-0 min-h-[16px] pb-7 text-xs font-bold'}>Filters</h4>
-      <div className={`flex gap-x-2`}>
-        {filters.map(({ title, accessor, options }) => (
-          <MultiSelect
-            key={keyFactory(title, filter?.[accessor])}
-            title={title}
-            selectedValues={filter?.[accessor] ?? []}
-            onSelect={onFilter(accessor)}
-            onClearSelect={onClearSelect(accessor)}
-            options={options}
-          />
-        ))}
-      </div>
-    </div>
+    <Filters
```

Comment:
Why the change to using React context and all these abstractions?

Reply:
In order to avoid props drilling, all childrens should have access to whole context.

Reply:
Before the change the props were passed to the multi select and now they are passed to the `Filters` component. What prop drilling?